core/src/main/java/org/kiwix/kiwixmobile/core/main/KiwixWebView.kt

```diff
@@ -83,12 +85,12 @@ open class KiwixWebView @SuppressLint("SetJavaScriptEnabled") constructor(
     clearCache(true)
     setWebViewClient(webViewClient)
     webChromeClient = KiwixWebChromeClient(callback, nonVideoView, videoView, this).apply {
-      setOnToggledFullscreen { fullscreen ->
-        setWindowVisibility(
-          if (fullscreen) View.SYSTEM_UI_FLAG_LOW_PROFILE else View.SYSTEM_UI_FLAG_VISIBLE
-        )
-        callback.onFullscreenVideoToggled(fullscreen)
-      }
+      setOnToggledFullscreen(object : ToggledFullscreenCallback {
+        override fun toggledFullscreen(fullscreen: Boolean) {
+          setWindowVisibility(fullscreen)
```

Comment:
You forgot to add `SYSTEM_UI_FLAG_LOW_PROFILE` and `SYSTEM_UI_FLAG_VISIBLE ` if it's fullscreen or not.